packages/react-i18n/src/i18n.ts

```diff
@@ -306,6 +306,19 @@ export default class I18n {
 
     return decimal.length === 0 ? DECIMAL_NOT_SUPPORTED : decimal;
   }
+
+  @memoize(
+    (
+      locale: string,
+      options: Intl.DateTimeFormatOptions & {style?: DateStyle} = {},
```

Comment:
I think the `DateStyle` helps a lot here... Maybe we can add more of these to accommodate more cases so that less devs use the `options` object.